var/spack/repos/builtin/packages/ginkgo/package.py

```diff
@@ -44,6 +44,9 @@ class Ginkgo(CMakePackage, CudaPackage, ROCmPackage):
         values=("Debug", "Release"),
     )
 
+    # make sure Debug mode is not used when optimizing
+    conflicts("build_type=Debug", when="+full_optimizations", msg="Use Release mode when optimizing")
```

Comment:
I think something more concise like this will be all we need moving forward for HiOp and/or ExaGO...